### PR TITLE
chore: update relative job positioning api

### PR DESCRIPTION
**Pull request by hamidzr**
## Description

- reorganize msgs used for adjusting priority, weight, and q position
- add ahead_of and behind_of actions for updating job order replacing the exposed q_position

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: proto/src/determined/job/v1/job.proto [hunk 1]
```diff
@@ -102,14 +102,14 @@ message QueueControl {
   };
   // Job id.
   string job_id = 1;
-  // The resource pool to perform an action on.
-  string source_resource_pool = 2;
   // The action to perform.
   oneof action {
+    // The desired job position in the queue in terms of another job.
+    string ahead_of = 2;
```
Comment (hamidzr) [inline proto/src/determined/job/v1/job.proto:108]: before and after ? 🤔

**Comment by hamidzr** [comment]
@eecsliu lmk if this lg to you. I think we can land this and separately implement the ahead_of and behind_of API parameters.

File: proto/src/determined/job/v1/job.proto (hunk 1 above)
Comment (eecsliu) [reply proto/src/determined/job/v1/job.proto:108]: is this expected to be a jobID?

**Review by eecsliu — APPROVED** [review]
Looks great! Appreciate the cleanup of everything.

File: proto/src/determined/job/v1/job.proto (hunk 1 above)
Comment (hamidzr) [reply proto/src/determined/job/v1/job.proto:108]: yes

**Pull request merged**